### ISSUE TITLE
Relative date queries as presets in collections module

### ISSUE THREAD

**Issue by derkork** [darktable-org/darktable#10239]
**Is your feature request related to a problem? Please describe.**

I often need to filter images by their date fields using relative queries (e.g. pictures i imported within the last 7 days, pictures I have modified within the last 48 hours). There are two built-in presets which can look for import dates within the last 24h and 30 days. However both of these are hard-coded to calculate the relative date when you select them and if I want to combine them with other filters I have no way of doing so and saving this as a preset.

**Describe the solution you'd like**

Ideally I would like to have some relative date calculations built into the query language. Basically i need a way of saving _in the last `x <time unit>`_ as a preset. I cannot think of a reason for supporting future days, so maybe something like:

```bash
change_timestamp: >= -3h   #finds all images changed within the last 3hours
change_timestamp: >= -30m  #changed  in the last 30 minutes
date_taken: <= -30d #taken more than 30 days ago
date_taken: [-180d, -90d] # taken between 180 and 90 days ago
```

So basically the parser that currently parses `yyyy:MM:dd` would need to have another regex to look for like `-[0-9]+[hmd]` and then calculate a relative date from that based on the current date but the general parsing and querying logic would remain the same. I think it is sufficient to support a single unit per query only (e.g. no mixing of hours, minutes and days). I also think that "month" and "year" as a unit should **not** be included because that would require more complex, calendar-based  computations catering for varying year and month lengths.

**Alternatives**

Currently there is no real alternative built-in. I cannot save a relative date as a preset, so every time I want to filter like that I would need to calculate the date by hand and insert it into the query.
